docs/update-from-client-side-helpers-to-server-side.md

```diff
@@ -0,0 +1,107 @@
++                 method: "POST",
++                 // use the "body" param to optionally pass additional order information
++                 // like product skus and quantities
++                 body: JSON.stringify({
++                   cart: [
++                     {
++                        sku: "YOUR_PRODUCT_STOCK_KEEPING_UNIT",
++                        quantity: "YOUR_PRODUCT_QUANTITY",
++                      }],
++                   }),
++                 })
++                 .then((response) => response.json())
++                 // return the PayPal Order ID
++                 .then((order) => order.id)
++               );
+            },
+            // Finalize the transaction
+            onApprove: function (data, actions) {
+-             return actions.order
+-               .capture()
+-               .then(function (orderData) {
+-                  // Successful capture!
+-                })
+-                .catch(function (err) {
+-                   // Failed capture
+-                });
++                return fetch("/my-server/patch-paypal-order", {
++                  method: "POST",
++                })
++                .then((response) => response.json())
++                .then(function (orderData) {
++                   // Successful capture!
++                })
++                .catch(function (err) {
++                   // Failed capture
++                });
+            },
+            onError: function (err) {
+              // For example, redirect to a specific error page
+              window.location.href = "/your-error-page-here";
+            }
+        })
+        .render("#paypal-button-container");
+    </script>
+  </body>
+</html>
+```
+
+- What is server-side code?
+
+  Server-side code runs securely on a web server and is typically used to communicate with APIs and Databases. Here's a list of common server-side languages used to make websites: Node.js, PHP, ASP.NET, Ruby, Java.
+
+- What should I do if I do not have the ability to run server-side code?
+
+  We recommend using one of PayPal's partners to host your website like Wix, GoDaddy, Shopify, and Big Commerce.
+
+## Best Practices for JS SDK Server-Side Integrations:
+
+1. CLIENT_SECRET should never be checked into git. We recommend passing this sensitive value to the web server at runtime as an environment variable. It's common to use a .env file that is ignored by git to load sensitive values like this secret.
+Examples:
+  * Node.js: `PAYPAL_CLIENT_SECRET=ABC_123_DOE_RAY_ME node server.js` and use `process.env` to read this value.
+  * Python: `PAYPAL_CLIENT_SECRET=ABC_123_DOE_RAY_ME my_cool_application.py` and use `os.environ` to read this value.
+
+2. The authorization token returned by `/v1/oauth2/token` api endpoint should never be passed to the browser. Keep this value in memory on the server-side and use it as the Authorization header for all other api calls.
```

Comment:
```suggestion
2. The client credentials auth token returned by `/v1/oauth2/token` api endpoint should never be passed to the browser. Keep this value in memory on the server-side and use it as the Authorization header for all other api calls.
```

Reply:
I think we should also add a Node.js code snippet here to provide more context with this point.

Node.js auth token example api call:

```js
const encodedClientCredentials = Buffer.from(`${client}:${secret}`).toString("base64");

const response = await fetch(`${apiBaseUrl}/v1/oauth2/token`, {
    method: "POST",
    body: "grant_type=client_credentials",
    headers: {
      Accept: "application/json",
      "Content-Type": "application/x-www-form-urlencoded",
      "Accept-Language": "en_US",
      Authorization: `Basic ${encodedClientCredentials}`,
    });
    
const data = await response.json();

// do not expose "data.access_token" to the browser
```